syntax = "proto3";

package pb;

service HttpClient {
	rpc GetURL(Request) returns (Response) {}
}

message Request {
	string url = 1;
	string host = 2;
	string domainName = 3;
	string proxy = 4;
	string useragent = 5;
	string requestMethod = 6;
	map<string, string> headers = 7;
	string scrapySpider = 8;
	string clientHello = 9;
}

message Response {
	map<string, string> headers = 1;
	bytes body = 2;
	string error = 3;
	string status = 4;
}
